test/unit/template_importer_test.rb

```diff
@@ -0,0 +1,106 @@
+      }.merge(opts))
+    end
+
+    setup do
+      @importer = importer
+    end
+
+    context 'get_default_branch' do
+      setup do
+        @repo = Struct.new(:branches).new([
+          Struct.new(:name).new('0.1-stable'),
+          Struct.new(:name).new('develop')
+        ])
+      end
+
+      test 'when on develop, returns develop' do
+        SETTINGS[:version].stubs(:tag).returns('develop')
+        assert_equal 'develop', @importer.get_default_branch(@repo)
+      end
+
+      test 'when branch exists, return it' do
+        SETTINGS[:version].stubs(:tag).returns('not_develop')
+        SETTINGS[:version].stubs(:short).returns('0.1')
+        assert_equal '0.1-stable', @importer.get_default_branch(@repo)
+      end
+
+      test 'when branch does not exist, use default branch' do
+        SETTINGS[:version].stubs(:tag).returns('not_develop')
+        SETTINGS[:version].stubs(:short).returns('0.2')
+        refute @importer.get_default_branch(@repo)
+      end
+    end
+
+    test 'metadata method extracts correct metadata' do
+      text = File.read(get_template('metadata1.erb'))
+      hash = @importer.parse_metadata(text)
+      assert_equal 'provision', hash['kind']
+      assert_equal 'Test Data', hash['name']
+      assert_equal 5, hash['oses'].size
+
+      text = File.read(get_template('metadata2.erb'))
+      hash = @importer.parse_metadata(text)
+      assert_equal 'ProvisioningTemplate', hash['model']
+    end
+
+    test 'import! loads templates into db' do
+      @importer.import!
+
+      # Check template was imported
+      ct = ProvisioningTemplate.find_by_name('FooBar Test Data')
+      assert ct.present?
+      assert_equal File.read(get_template('metadata1.erb')), ct.template
+      assert_equal 1, ct.operatingsystems.size
+      assert_equal Operatingsystem.find_by_title('Ubuntu 10.10').id, ct.operatingsystems.first.id
+
+      # Check model-based template was imported
+      ct = ProvisioningTemplate.find_by_name('FooBar Model-based Test')
+      assert ct.present?
+      assert_equal File.read(get_template('metadata2.erb')), ct.template
+      assert_equal 1, ct.operatingsystems.size
+      assert_equal Operatingsystem.find_by_title('Ubuntu 10.10').id, ct.operatingsystems.first.id
+
+      # Check snippet was imported
+      sp = ProvisioningTemplate.find_by_name('FooBar Test Snippet')
+      assert sp.present?
+      assert_equal File.read(get_template('snippet1.erb')), sp.template
+
+      # Check ptable was imported
+      pt = Ptable.find_by_name('FooBar Test Ptable')
+      assert pt.present?
+      assert_equal File.read(get_template('ptable1.erb')), pt.layout
+      assert_equal 'Debian', pt.os_family
+    end
+
+    test 'purge! removes expected template' do
+      FactoryGirl.create(:provisioning_template, :name => 'Community delete_me')
+      FactoryGirl.create(:provisioning_template, :name => 'keep_me')
+      @importer.purge!
+      ct = ProvisioningTemplate.find_by_name('delete_me')
```

Comment:
This can never fail - you're creating "Community delete_me" but testing for presence of "delete_me" - you need to update the name here to include the prefix